game-core/src/main/java/games/strategy/engine/framework/ArgParsingHelper.java

```diff
@@ -0,0 +1,51 @@
+package games.strategy.engine.framework;
+
+import java.util.Arrays;
+import java.util.Properties;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+
+/**
+ * A helper class for parsing args. Looks for args beginning with "-P" and will
+ * return them as part of a {@code Properties} object.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class ArgParsingHelper {
+  @VisibleForTesting
+  static final String TRIPLEA_PROPERTY_PREFIX = "P";
+
+  public static Properties getTripleaProperties(final String... args) {
+    final Options options = getOptions();
+    final CommandLineParser parser = new DefaultParser();
+    try {
+      final CommandLine cli = parser.parse(options, args);
+      return cli.getOptionProperties(TRIPLEA_PROPERTY_PREFIX);
+    } catch (ParseException e) {
```

Comment:
Getting this into checkstyle would be good. Otherwise I think we should skip reviewing for this unless we are looking at object state. To follow up in more detail: https://github.com/triplea-game/triplea/issues/3660

In the meantime I'll fix this.

Reply:
Eagle eye @RoiEXLab , but to be sure, unless we can automate checking for final, I don't think we should be reviewing for it unless it is in the context of immutable design.

Reply:
Indeed, this can be automated with Checkstyle.  See #3662.